http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/utils/NettyUtils.java

```diff
@@ -131,4 +131,18 @@ public static void doInEventLoop(EventExecutor eventExecutor, Runnable runnable,
             promise.setFailure(e);
         }
     }
+
+    /**
+     * Wait interruptibly for the promise to complete.
+     *
+     * @param promise the promise to complete
+     */
+    public static void syncInterruptibly(Future<?> promise) {
```

Comment:
This isn't that much different than netty's `awaitUninterruptibly()`. Any reason we care about re-throwing the interrupt exception if we're just going to log it?

Reply:
Per Javadoc of `awaitUniterrupttibly`,
> Waits for this future to be completed within the specified time limit without interruption.  This method catches an {@link InterruptedException} and discards it silently.

it doesn't re-interrupt the thread and that's why I created this method to re-interrupt the thread.

Reply:
This does re-interrupt the thread, if you look at the implementation.

Reply:
Ohh yeah, it does. I will update. I was looking at the implementation of an overloaded method of `awaitUninterruptibly` and from first look, it looks like it doesn't re-interrupt the method.
```
    @Override
    public boolean awaitUninterruptibly(long timeout, TimeUnit unit) {
        try {
            return await0(unit.toNanos(timeout), false);
        } catch (InterruptedException e) {
            // Should not be raised at all.
            throw new InternalError();
        }
    }
```